frontend/src/pages/Common/HistoryList.js

```diff
@@ -18,7 +19,7 @@ const styles = {
 
 export default function HistoryList({ events, nEventsTotal, hasMore, isLoading, getUserDisplayname }) {
   const eventItems = events.map((event, index) => {
-    const eventTime = event.getIn(["businessEvent", "time"]);
+    const eventTime = event.businessEvent.time;
```

Comment:
Please check if the event is valid by evaluating `businessEvent` and `snapshot`

Reply:
done